component/src/main/java/io/siddhi/extension/execution/reorder/AlphaKSlackExtension.java

```diff
@@ -205,218 +199,202 @@
-        previousError = (Double) map.get("previousError");
-        kp = (Double) map.get("kp");
-        kd = (Double) map.get("kd");
-        primaryTreeMap = (TreeMap) map.get("primaryTreeMap");
-        secondaryTreeMap = (TreeMap) map.get("secondaryTreeMap");
-        dataItemList = (List) map.get("dataItemList");
-        timestampList = (List) map.get("timestampList");
+    public void stop() {
+        //Do nothing
     }
 
     @Override
     protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processor nextProcessor,
-                           StreamEventCloner streamEventCloner,
-                           ComplexEventPopulater complexEventPopulater) {
-        ComplexEventChunk<StreamEvent> complexEventChunk = new ComplexEventChunk<StreamEvent>(false);
-        try {
-            lock.lock();
-            while (streamEventChunk.hasNext()) {
-                StreamEvent event = streamEventChunk.next();
-
-                if (event.getType() != ComplexEvent.Type.TIMER) {
-                    streamEventChunk.remove();
-                    long timestamp = (Long) timestampExecutor.execute(event);
-                    timestampList.add(timestamp);
-                    double correlationField;
-                    switch (attributeExpressionExecutors[1].getReturnType()) {
-                        case INT:
-                            correlationField = (Integer) correlationFieldExecutor.execute(event);
-                            break;
-                        case LONG:
-                            correlationField = (Long) correlationFieldExecutor.execute(event);
-                            break;
-                        case FLOAT:
-                            correlationField = (Float) correlationFieldExecutor.execute(event);
-                            break;
-                        case DOUBLE:
-                            correlationField = (Double) correlationFieldExecutor.execute(event);
-                            break;
-                        default:
-                            //will not occur at all
-                            correlationField = 0.0;
-
-                    }
-                    dataItemList.add(correlationField);
-                    if (discardFlag) {
-                        if (timestamp < lastSentTimestamp) {
-                            continue;
-                        }
-                    }
-
-                    if (timerFlag) {
-                        timerFlag = false;
-                        lastScheduledTimestamp = lastScheduledTimestamp + timerDuration;
-                        scheduler.notifyAt(lastScheduledTimestamp);
-                    }
+                           StreamEventCloner streamEventCloner, ComplexEventPopulater complexEventPopulater,
+                           AlphaKSlackState state) {
+        synchronized (state) {
+            ComplexEventChunk<StreamEvent> complexEventChunk = new ComplexEventChunk<StreamEvent>(false);
```

Comment:
Can you test with new ComplexEventChunk<StreamEvent>(true); 
if all test passes please make it true

Reply:
Both `KSlackExtension` and `AlphaKSlackExtension` passes with `ComplexEventChunk(true)`. Fixed this in commit 2e2e476.